gensim/similarities/docsim.py

```diff
@@ -653,6 +658,9 @@ def get_similarities(self, query):
         if result.shape[1] == 1 and not is_corpus:
             # for queries of one document, return a 1d array
             result = result.toarray().flatten()
+        elif self.maintain_sparsity:
+            # avoid converting to dense array if maintaining sparsity
+            result = result.T
```

Comment:
The line above, `result.toarray().flatten()`, will still densify the output. So you'll get sometimes dense, sometimes sparse, depending which `if` branch is hit.

I don't know your use case, but isn't it better to do this `maintain_sparsity` check as the first thing, so that the output is *always* sparse? The API seems cleaner that way.